### ISSUE TITLE
Rich box with multilingual

### ISSUE THREAD

**Issue by mustafaabduljalil** [thedevdojo/voyager#4152]
## Version information

 - Laravel: v 5.8
 - Voyager: v 1.2
 - PHP:7.1.3
 - Database: [type] [version] (e.g. MySQL 8.0)

## Description

Rich box dose not save changes i use en and ar languages but give me this error 
Undefined index: en and i tried all issues that on issues tabs in voyagers of same problem
please i need help

## Steps To Reproduce
this is my config/voyager.php section:
'multilingual' => [ 'enabled' => true, 'default' => 'it', 'locales' => [ 'en', 'ar' ], ],
on my models i put use TCG\Voyager\Traits\Translatable; and use Translatable; into class.
I also set protected $translatable = ['title', 'description']; to set what fields need to be translated.

**Comment by marijnz0r** [comment]
Confirmed, I'm experiencing the same problem as well.
* Laravel 5.5
* Voyager 1.2
* PHP 7.2.10

**Comment by marijnz0r** [comment]
I see this is already mentioned in https://github.com/the-control-group/voyager/issues/4103

**Comment by NurdaKz** [comment]
You can try this #4077

**Comment by mustafaabduljalil** [comment]
@NurdaKz I tried this solution but not works for me !!

**Comment by mentAl-maZe** [comment]
If you've got node installed you can use this little script i made
Refer to my comment in [#4072](https://github.com/the-control-group/voyager/issues/4072#issuecomment-494380285)

**Comment by really4you** [comment]
#4129

**Comment by emptynick** [comment]
Fixed in https://github.com/the-control-group/voyager/pull/4157
Also duplicate of https://github.com/the-control-group/voyager/issues/4072

**Comment by selusi** [comment]
I tried all the following fixes (#4072, #4152 #4157):
`inpUsr = $(_inp).parent().find(_this.settings.editing ? '.form-control' : '');` 
and
`inpUsr = _inp.nextAll(_this.settings.editing ? '.form-control' : '');`

Both fixes work well on pages and posts forms, but if I want to change the menu items nothing works.
So I find a more complex fix but it works well on all forms.
The first step is to create e view for the rich text hidden field: 
input-hidden-rich-text.blade.php:
```
@if ($isModelTranslatable)
    <input type="hidden"
           data-i18n="true"
	   class="hiddeRichText"
           name="{{ $_field_name }}_i18n"
           id="{{ $_field_name }}_i18n"
           value="{{ $_field_trans }}">
@endif 
```
The class '**hiddeRichText**' is used to recognize that the field is related to a rich text field.

obviously I have changed the inclusion on the edit-add.blade.php file:
```
 @include('voyager::multilingual.input-hidden-rich-text', [
                                '_field_name'  => 'body',
                                '_field_trans' => get_field_translations($dataTypeContent, 'body')
                            ])
```
Then I modified the file multilingual.js:
I added the method **isHiddenRichText**
```
 /**
     * Chek if the field is a rich text field edited with tinymce.
	 * By referencing to specific view: vendor\tcg\voyager\resources\views\multilingual\input-hidden-rich-text.blade.php
     *
     * @return boolean.
     */
	isHiddenRichText: function isHiddenRichText(inp){		
		var setSelector = $.trim(inp.prop('name'));		
		if( $('#'+setSelector).hasClass('hiddeRichText') ){
			return true;
		}
		return false;
	},
```
then I modify the functions: **updateInputCache** and **loadLang**:
```
updateInputCache: function updateInputCache(inp) {
		var	_this = this,
		isRichText = _this.isHiddenRichText(inp),
		inpUsr = inp.data('inpUsr'),
		$_val = $(inpUsr).val(), // current value
		$_data = {}; // Create new data

		/* If the field is edited by tinymce, It retrive the value from the istance of the editor */
		if( isRichText ){		
			var $_mce = tinymce.get('richtext'+inp.prop('name').replace("_i18n", "") );
			$_val = $_mce.getContent();
		} 

		this.langSelectors.each(function (i, btn) {
			var lang = btn.id;			
			$_data[lang] = _this.locale == lang ? $_val : inp.data(lang);
		});
		inp.val(JSON.stringify($_data));
		inp.data(this.locale, $_val); // Update single key Mem
	},

    /**
     * Load input translation
     */
    loadLang: function loadLang(inp, lang) {
      var inpUsr = inp.data("inpUsr"),
          _val = inp.data(lang);

      if (!this.settings.editing) {
         inpUsr.text(_val);
      } else {
	/* Retrive the value of richtext box */
        var _mce = tinymce.get('richtext' + inp.prop('name').replace("_i18n", ""));
        /* If the field is edited by tinymce, set the content of the editor */
		if (this.isHiddenRichText(inp)) {
                   _mce.setContent(_val);
                } else {
                   inpUsr.val(_val);
               }
        }
    }
```
All files 

[fix.zip](https://github.com/the-control-group/voyager/files/3258834/fix.zip)

Of course it is possible to improve the code but for the moment the important thing is that it works.
Laravel 5.8
PHP 7.2.8
